### DISCUSSION TITLE
Matter Endpoint

### DISCUSSION THREAD

**Discussion by Nezaemmy** [espressif/arduino-esp32#10672] [Q&A]
Hello @SuGlider do you have a plan to add API in Matter library to be able to change the number of endpoints the Matter library can support?
in esp-idf I found this option ```CONFIG_ESP_MATTER_MAX_DYNAMIC_ENDPOINT_COUNT=```.
Right now the library support 15 endpoints.

**Comment by SuGlider** [comment]
> Hello @SuGlider do you have a plan to add API in Matter library to be able to change the number of endpoints the Matter library can support? in esp-idf I found this option `CONFIG_ESP_MATTER_MAX_DYNAMIC_ENDPOINT_COUNT=`. Right now the library support 15 endpoints.

This can be achieved by building the project using Arduino as Component and changing the ConnectHomeIP or ESP Matter settings directly in its source code.

No plans for changing CHIP /  ESP Matter settings.

**Reply by SuGlider** [reply]
There is a trade off about number and flash/RAM footprint.
https://github.com/espressif/esp-matter/blob/main/docs/en/optimizations.rst#624-adjust-the-dynamic-endpoint-count
